lib/manager/npm/post-update/index.ts

```diff
@@ -156,7 +159,7 @@ export async function writeExistingFiles(
         logger.warn({ npmrcFilename, err }, 'Error writing .npmrc');
       }
     }
-    if (packageFile.yarnrc) {
+    if (packageFile.managerData?.yarnrc) {
```

Comment:
If you redefine `AdditionalPackageFiles` interface as

```typescript
export interface AdditionalPackageFiles {
  npm?: Partial<PackageFile<NpmManagerData>>[];
}
```

it will simplify the type checking/type guards below.